File: tests/fixtures/python/slop_restate.py
class HitCounter:
    def __init__(self):
        self.count = 0

    def record_hit(self):
        # increment the count
        self.count += 1

    def record_batch(self, n):
        self.count += n  # increment the count


# expect-line: 6 SLOP042
# expect-line: 10 SLOP042

File: tests/fixtures/rust/slop_restate.rs
fn record_hit(counter: &mut i32) {
    // increment the counter
    *counter += 1;
}

fn record_batch(counter: &mut i32, n: i32) {
    *counter += n; // increment the counter
}

fn main() {
    let mut hits = 0;
    record_hit(&mut hits);
    record_batch(&mut hits, 3);
}

// expect-line: 2 SLOP042
// expect-line: 7 SLOP042
